reader_test.go

```diff
@@ -298,169 +279,139 @@ func (s *MySuite) TestNestedOffsetDecode(c *C) {
 
 	expected := newInvalidDatabaseError("the MaxMind DB contains invalid metadata")
-	c.Assert(err, DeepEquals, expected)
+	assert.True(t, assert.ObjectsAreEqual(err, expected))
 }
 
-func (s *MySuite) TestMissingDatabase(c *C) {
+func TestMissingDatabase(t *testing.T) {
 	reader, err := Open("file-does-not-exist.mmdb")
-	if reader != nil {
-		c.Log("received reader when doing lookups on DB that doesn't exist")
-		c.Fail()
-	}
-	c.Assert(err, ErrorMatches, "open file-does-not-exist.mmdb.*")
+	assert.Nil(t, reader, "received reader when doing lookups on DB that doesn't exist")
+	assert.Regexp(t, "open file-does-not-exist.mmdb.*", err)
 }
 
-func (s *MySuite) TestNonDatabase(c *C) {
+func TestNonDatabase(t *testing.T) {
 	reader, err := Open("README.md")
-	if reader != nil {
-		c.Log("received reader when doing lookups on DB that doesn't exist")
-		c.Fail()
-	}
-	c.Assert(err.Error(), Equals, "error opening database: invalid MaxMind DB file")
+	assert.Nil(t, reader, "received reader when doing lookups on DB that doesn't exist")
+	assert.Equal(t, err.Error(), "error opening database: invalid MaxMind DB file")
 }
 
-func (s *MySuite) TestDecodingToNonPointer(c *C) {
+func TestDecodingToNonPointer(t *testing.T) {
 	reader, _ := Open("test-data/test-data/MaxMind-DB-test-decoder.mmdb")
 
 	var recordInterface interface{}
 	err := reader.Lookup(net.ParseIP("::1.1.1.0"), recordInterface)
-	c.Assert(err.Error(), Equals, "result param must be a pointer")
-	if err = reader.Close(); err != nil {
-		c.Assert(err, nil, "no error on close")
-	}
+	assert.Equal(t, err.Error(), "result param must be a pointer")
+	assert.Nil(t, reader.Close(), "error on close")
 }
 
-func (s *MySuite) TestNilLookup(c *C) {
+func TestNilLookup(t *testing.T) {
 	reader, _ := Open("test-data/test-data/MaxMind-DB-test-decoder.mmdb")
 
 	var recordInterface interface{}
 	err := reader.Lookup(nil, recordInterface)
-	c.Assert(err.Error(), Equals, "ipAddress passed to Lookup cannot be nil")
-	if err = reader.Close(); err != nil {
-		c.Assert(err, nil, "no error on close")
-	}
+	assert.Equal(t, err.Error(), "ipAddress passed to Lookup cannot be nil")
+	assert.Nil(t, reader.Close(), "error on close")
 }
 
-func checkMetadata(c *C, reader *Reader, ipVersion uint, recordSize uint) {
+func checkMetadata(t *testing.T, reader *Reader, ipVersion uint, recordSize uint) {
 	metadata := reader.Metadata
 
-	c.Assert(metadata.BinaryFormatMajorVersion, Equals, uint(2))
+	assert.Equal(t, metadata.BinaryFormatMajorVersion, uint(2))
 
-	c.Assert(metadata.BinaryFormatMinorVersion, Equals, uint(0))
-	c.Assert(metadata.BuildEpoch, FitsTypeOf, uint(0))
-	c.Assert(metadata.DatabaseType, Equals, "Test")
+	assert.Equal(t, metadata.BinaryFormatMinorVersion, uint(0))
+	assert.IsType(t, uint(0), metadata.BuildEpoch)
+	assert.Equal(t, metadata.DatabaseType, "Test")
 
-	c.Assert(metadata.Description, DeepEquals,
+	assert.ObjectsAreEqual( metadata.Description,
```

Comment:
`assert.ObjectsAreEqual` does not do any test assertions so this will never fail.